src/main/java/net/openhft/chronicle/queue/impl/single/SingleChronicleQueue.java

```diff
@@ -1079,54 +1092,15 @@ public SingleChronicleQueueStore acquire(int cycle, boolean createIfAbsent) {
             }
         }
 
-        private synchronized void headerRecovery(final SingleChronicleQueue that, final MappedBytes mappedBytes, final AbstractWire wire, final Bytes<?> bytes, final int cycle) throws IOException, TimeoutException {
-
-            try {
-                final RandomAccessFile raf = mappedBytes.mappedFile().raf();
-                FileLock fileLock = ReentrantFileLock.tryLock(mappedBytes.mappedFile().file(), raf.getChannel());
-                // Throw if we couldn't get the lock
-                if (fileLock == null) {
-                    throw new IOException("Couldn't get lock to recover header file");
-                }
-
-                try {
-
-                    final int header = bytes.readVolatileInt(0);
-                    if (isReady(header))
-                        return;
-
-                    // we hold the file lock so are going to force recovery
-                    if (!bytes.compareAndSwapInt(0, header, 0)) {
-                        Jvm.warn().on(getClass(), "failed to recover.");
-                        throw new StreamCorruptedException("failed to recover.˚");
-                    }
-
-                    if (!wire.writeFirstHeader()) {
-                        Jvm.warn().on(getClass(), "failed to recover.");
-                        throw new StreamCorruptedException("failed to recover.˚");
-                    }
+        // Rename un-acquirable segment file to make sure no data is lost to try and recreate the segment
+        private boolean backupCycleFile(File cycleFile) {
+            File cycleFileDiscard = new File(cycleFile.getParentFile(), cycleFile.getName() + DISCARD_FILE_SUFFIX);
+            boolean success = cycleFile.renameTo(cycleFileDiscard);
```

Comment:
Is it possible that multiple processes get here and think they moved the existing file (i.e is the move atomic), and would that cause chaos later if they both attempted to create the replacement file? I notice [Files.move](https://docs.oracle.com/javase/8/docs/api/java/nio/file/Files.html#move-java.nio.file.Path-java.nio.file.Path-java.nio.file.CopyOption...-) allows you to specify an atomic move (and define the behaviour when an existing file is there), not sure if that would be more appropriate here.

Reply:
Only one of these process succeeds and this one gets to recreate the file from scratch - other ones retry opening this file without createIfAbsent flag. I find the behavior of File.renameTo() more consistent than the described atomic move, but when we rename file in the current directory it should almost always be atomic yes or no operation.